numpy/lib/recfunctions.py

```diff
@@ -948,6 +965,8 @@ def structured_to_unstructured(arr, dtype=None, copy=False, casting='unsafe'):
 
     fields = _get_fields_and_offsets(arr.dtype)
     n_fields = len(fields)
+    if n_fields == 0:
+        raise ValueError("dtype must have at least one field")
```

Comment:
I don't think this is a necessary requirement - if you used a for loop instead of zip below, you would not need a special case. What you're working around here is that `zip(*x)` does not generalize well, which is an argument against using it.